kernel/src/syscall/getsid.rs

```diff
@@ -9,23 +9,19 @@ use crate::{
 pub fn sys_getsid(pid: Pid, ctx: &Context) -> Result<SyscallReturn> {
     debug!("pid = {}", pid);
 
-    let session = ctx.process.session().unwrap();
-    let sid = session.sid();
-
     if pid == 0 {
-        return Ok(SyscallReturn::Return(sid as _));
+        return Ok(SyscallReturn::Return(ctx.process.sid() as _));
     }
 
-    let Some(process) = process_table::get_process(pid) else {
-        return_errno_with_message!(Errno::ESRCH, "the process does not exist")
-    };
+    let process = process_table::get_process(pid).ok_or(Error::with_message(
+        Errno::ESRCH,
+        "the process to get the SID does not exist",
+    ))?;
 
-    if !Arc::ptr_eq(&session, &process.session().unwrap()) {
-        return_errno_with_message!(
-            Errno::EPERM,
-            "the process and current process does not belong to the same session"
-        );
-    }
+    // The Linux kernel implementation does not check whether `process` is in the same session as
+    // the current process. We follow it.
 
-    Ok(SyscallReturn::Return(sid as _))
+    // FIXME: If we call `sid()` without holding the process table lock, it may return zero if the
+    // process is reaped at the same time. Is this behavior correct?
```

Comment:
I'm puzzled about why the condition mentioned in this FIXME would occur. If the process has been reaped, it should already be a zombie process, meaning the syscall result cannot be returned to user space. Consequently, no one would be concerned about the syscall result.

Reply:
`getsid(pid)` can be used to get the SID of a process that is not currently running on the CPU. So it can be done concurrently. I'm not sure what you mean by "the syscall result cannot be returned to user space"?

Reply:
Thank you for the explanation; I had overlooked the fact that the PID can refer to a process other than the current process.